packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -265,22 +293,38 @@ async def archive_dir(
 
     The **exclude_patterns** is a set of patterns created using
     Unix shell-style wildcards to exclude files and directories.
+
+    destination: Path deleted if errors
+
+    ::raise ArchiveError
     """
-    with non_blocking_process_pool_executor(max_workers=1) as pool:
-        add_to_archive_error: Union[
-            None, Exception
-        ] = await asyncio.get_event_loop().run_in_executor(
-            pool,
-            _add_to_archive,
-            dir_to_compress,
-            destination,
-            compress,
-            store_relative_path,
-            exclude_patterns,
-        )
-
-        if isinstance(add_to_archive_error, Exception):
-            raise add_to_archive_error
+    with non_blocking_process_pool_executor(max_workers=1) as process_pool:
```

Comment:
if we use only 1 worker does it make sense to use the process pool executor?

Reply:
yep. Discussed offline with SAN